### ISSUE TITLE
1.26 Release planning

### ISSUE THREAD

**Issue by cpackham** [crosstool-ng/crosstool-ng#1991]
Blocking issues: ~#1989~, ~#1990~
Nice to have: #1569, ~#2002~

- [x] Tag 1.26.0-rc1
- - [x] Upload 1.26.0-rc1 tarball to github
- - [x] Upload 1.26.0-rc1 tarball to crosstool-ng.org?
- [x] Run `build-all`
- [x] Tag 1.26.0-rc2 (if needed)
- - [x] Upload 1.26.0-rc2 tarball to github
- - [x] Upload 1.26.0-rc2 tarball to crosstool-ng.org?
- [x] Tag 1.26.0
- - [x] Upload 1.26.0 tarball to github
- - [x] Upload 1.26.0 tarball to crosstool-ng.org
- [x] Release announcement
- - [x] crossgcc ML 
- - [x] crosstool-ng website

**Comment by nolange** [comment]
This one seems rather serious to me: #1973

Given that newlib 4.3.0 fails to build with old GCC versions when those patches are missing

**Comment by modbw** [comment]
I have build all samples on Ubuntu 20.04. 9 of 118 samples failed.  These are:

```
arm-multilib-linux-uclibcgnueabi
i686-w64-mingw32_nios2-spico-elf
moxie-unknown-moxiebox
powerpc-unknown-linux-uclibc_m68k-unknown-uclinux-uclibc
x86_64-multilib-linux-uclibc_moxie-unknown-moxiebox
x86_64-multilib-linux-uclibc_powerpc-unknown-elf
x86_64-unknown-linux-gnu_arm-picolibc-eabi
x86_64-w64-mingw32_arm-cortexa9_neon-linux-gnueabihf
xtensa-fsf-elf
```
It took about 9 hours on my build server to run the test using [this](https://gist.github.com/modbw/e6454472642dc3c949967bb00446eb34) script which runs multiple builds in parallel.

[arm-multilib-linux-uclibcgnueabi.build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/12247429/arm-multilib-linux-uclibcgnueabi.build.log.gz)
[i686-w64-mingw32_nios2-spico-elf.build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/12247430/i686-w64-mingw32_nios2-spico-elf.build.log.gz)
[moxie-unknown-moxiebox.build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/12247431/moxie-unknown-moxiebox.build.log.gz)
[powerpc-unknown-linux-uclibc_m68k-unknown-uclinux-uclibc.build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/12247432/powerpc-unknown-linux-uclibc_m68k-unknown-uclinux-uclibc.build.log.gz)
[x86_64-multilib-linux-uclibc_moxie-unknown-moxiebox.build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/12247433/x86_64-multilib-linux-uclibc_moxie-unknown-moxiebox.build.log.gz)
[x86_64-multilib-linux-uclibc_powerpc-unknown-elf.build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/12247434/x86_64-multilib-linux-uclibc_powerpc-unknown-elf.build.log.gz)
[x86_64-unknown-linux-gnu_arm-picolibc-eabi.build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/12247435/x86_64-unknown-linux-gnu_arm-picolibc-eabi.build.log.gz)
[x86_64-w64-mingw32_arm-cortexa9_neon-linux-gnueabihf.build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/12247436/x86_64-w64-mingw32_arm-cortexa9_neon-linux-gnueabihf.build.log.gz)
[xtensa-fsf-elf.build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/12247439/xtensa-fsf-elf.build.log.gz)

**Comment by cpackham** [comment]
I've posted a PR that I think should sort the uclibc issues.

I'm tempted to drop xtensa-fsf-elf It's been marked as since before the last release and uses custom sources that haven't been updated in some time (the upstream esp32 support does seem to be moving forward). Naively trying to use the upstream versions doesn't seem to resolve the build issue.

I haven't looked at any of the other failures yet.

**Comment by cpackham** [comment]
@modbw I think for the canadian-cross (e.g. x86_64-unknown-linux-gnu,arm-picolibc-eabi) builds you missed a step to have the first toolchain in the path so you get failures like `[ERROR]  Missing: 'x86_64-unknown-linux-gnu-ar' or 'x86_64-unknown-linux-gnu-ar' or 'ar' : either needed!`

**Comment by cpackham** [comment]
The moxiebox failure seems to be with newlib 4.3. Changing to 4.2 it builds fine.

**Comment by modbw** [comment]
@cpackham I have updated my script to setup PATH for canadian cross builds. After rerunning the tests I see still 8 toolchains failing but with different errors now. All results are with crosstool-ng-1.25.0-217-gfd04f436 on Ubuntu 20.04

- [aarch64-rpi3-linux-gnu.build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/12341489/aarch64-rpi3-linux-gnu.build.log.gz)
`[ERROR]      configure: error: use a compatible toolchain or configure with --disable-mathvec (this results in incomplete ABI).` Caused by 2ef5c62c6884a1dd7e9600f2a965bb0ea9a077fb. Going back to glibc 2.37 fixes this issue
- [aarch64-rpi4-linux-gnu.build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/12341491/aarch64-rpi4-linux-gnu.build.log.gz) 
`[ERROR]      configure: error: use a compatible toolchain or configure with --disable-mathvec (this results in incomplete ABI).` Caused by 2ef5c62c6884a1dd7e9600f2a965bb0ea9a077fb. Going back to glibc 2.37 fixes this issue
- [i686-w64-mingw32_nios2-spico-elf.build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/12341492/i686-w64-mingw32_nios2-spico-elf.build.log.gz)
`Linking error with _imp__tputs` This may be fixed in https://github.com/crosstool-ng/crosstool-ng/commit/fab9efd8fad5c9a1b6a85816817f406c91702c56
- [moxie-unknown-moxiebox.build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/12341493/moxie-unknown-moxiebox.build.log.gz)
`Fails in Installing moxiebox runtime and VM`
- [x86_64-multilib-linux-uclibc_moxie-unknown-moxiebox.build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/12341494/x86_64-multilib-linux-uclibc_moxie-unknown-moxiebox.build.log.gz)
`ISL for host fails`
- [x86_64-multilib-linux-uclibc_powerpc-unknown-elf.build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/12341495/x86_64-multilib-linux-uclibc_powerpc-unknown-elf.build.log.gz)
`ISL for host fails`
- [x86_64-w64-mingw32_arm-cortexa9_neon-linux-gnueabihf.build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/12341496/x86_64-w64-mingw32_arm-cortexa9_neon-linux-gnueabihf.build.log.gz)
`Linking error with _imp__tputs` This may be fixed in https://github.com/crosstool-ng/crosstool-ng/commit/fab9efd8fad5c9a1b6a85816817f406c91702c56
- [xtensa-fsf-elf.build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/12341497/xtensa-fsf-elf.build.log.gz)
`error: 'struct _reent' has no member named '_atexit'`

**Comment by cpackham** [comment]
The rpi failures I think are related to the following from the glibc release notes:
```
* Vector math library libmvec support has been added to AArch64.  It
  requires GCC version >= 10.1.0.  It can be disabled via
  "--disable-mathvec", however that is not a supported configuration as
  it changes the ABI. The symbol names follow the AArch64 vector ABI,
  they are declared in math.h and have to be called manually at this point.
```
we should be satisfying that requirement but perhaps the configure script isn't getting it right

**Comment by cpackham** [comment]
#2023 should take care of the RPI stuff. Not convinced it's the best fix but it seems to work

**Comment by johan13** [comment]
It's great to see a new release is on its way. Thank you for the effort you put in!

I tried upgrading the three toolchains I have previously built on 1.25 to 1.26-rc1 and did not encounter any issues.

**Comment by cpackham** [comment]
ISL failure is
```
[ALL  ]      CXXLD    isl_test2
[ALL  ]      CXXLD    isl_test_cpp 
[ALL  ]    /home/ctng/x-tools/x86_64-multilib-linux-uclibc/lib/gcc/x86_64-multilib-linux-uclibc/13.2.0/../../../../x86_64-multilib-linux-uclibc/bin/ld.bfd: isl_test2.o: non-canonical reference to canonical protected function `__pthread_key_create' in /home/ctng/x-tools/x86_64-multilib-linux-uclibc/x86_64-multilib-linux-uclibc/sysroot/lib64/libc.so.1
[ALL  ]    /home/ctng/x-tools/x86_64-multilib-linux-uclibc/lib/gcc/x86_64-multilib-linux-uclibc/13.2.0/../../../../x86_64-multilib-linux-uclibc/bin/ld.bfd: failed to set dynamic section sizes: bad value
[ERROR]    collect2: error: ld returned 1 exit status
```

I don't speak enough C++ to know idea what that means

**Comment by nolange** [comment]
Don't think it has anything to do with C++, the linker complains that `__pthread_key_create` is defined/used with different visibility (ELF symbol stuff).
https://sourceware.org/bugzilla/show_bug.cgi?id=28875

**Comment by cpackham** [comment]
As a test I tried another canadian build with glibc instead of uclibc and ISL built fine.

**Comment by cpackham** [comment]
I've posted a potential fix for the ISL issue as #2027. Again I'm not sure it's right but it makes the problem go away.

**Comment by cpackham** [comment]
#2028 should fix the xtensa-fsf-elf build

**Comment by cpackham** [comment]
So as far as I know moxiebox is the only failing build. Seems to be OK with newlib 4.2 so something in between 4.2 and 4.3 has changed and causes those build failures. I think for 1.26.0 I might just update the config to pin the newlib version back certainly if no fix presents itself soon that'll be the option I'll go for.

@modbw do you have bandwidth to do another round of builds?

**Comment by modbw** [comment]
@cpackham I just started a build. Expect the results tomorrow at 9:00 CEST

**Comment by modbw** [comment]
@cpackham One sample failed to build on [crosstool-ng-1.26.0-rc2](https://github.com/crosstool-ng/crosstool-ng/tree/crosstool-ng-1.26.0-rc2)

- moxie-unknown-moxiebox
`Fails in Installing moxiebox runtime and VM`
[moxie-unknown-moxiebox.build.log.gz](https://github.com/crosstool-ng/crosstool-ng/files/12656320/moxie-unknown-moxiebox.build.log.gz)
```
[CFG  ]    /home/XXXXXX/tmp/ctng/x-tools-067-moxie-unknown-moxiebox/moxie-unknown-moxiebox/lib/gcc/moxie-unknown-moxiebox/13.2.0/../../../../moxie-unknown-moxiebox/bin/ld: /home/XXXXXX/tmp/ctng/x-tools-067-moxie-unknown-moxiebox/moxie-unknown-moxiebox/moxie-unknown-moxiebox/lib/libc.a(libc_a-closer.o): in function `_close_r':
[CFG  ]    /home/XXXXXX/tmp/ctng/build-067-moxie-unknown-moxiebox/.build/moxie-unknown-moxiebox/src/newlib/newlib/libc/reent/closer.c:47: undefined reference to `_close'
[CFG  ]    /home/XXXXXX/tmp/ctng/x-tools-067-moxie-unknown-moxiebox/moxie-unknown-moxiebox/lib/gcc/moxie-unknown-moxiebox/13.2.0/../../../../moxie-unknown-moxiebox/bin/ld: /home/XXXXXX/tmp/ctng/x-tools-067-moxie-unknown-moxiebox/moxie-unknown-moxiebox/moxie-unknown-moxiebox/lib/libc.a(libc_a-lseekr.o): in function `_lseek_r':
[CFG  ]    /home/XXXXXX/tmp/ctng/build-067-moxie-unknown-moxiebox/.build/moxie-unknown-moxiebox/src/newlib/newlib/libc/reent/lseekr.c:49: undefined reference to `_lseek'
[CFG  ]    /home/XXXXXX/tmp/ctng/x-tools-067-moxie-unknown-moxiebox/moxie-unknown-moxiebox/lib/gcc/moxie-unknown-moxiebox/13.2.0/../../../../moxie-unknown-moxiebox/bin/ld: /home/XXXXXX/tmp/ctng/x-tools-067-moxie-unknown-moxiebox/moxie-unknown-moxiebox/moxie-unknown-moxiebox/lib/libc.a(libc_a-readr.o): in function `_read_r':
[CFG  ]    /home/XXXXXX/tmp/ctng/build-067-moxie-unknown-moxiebox/.build/moxie-unknown-moxiebox/src/newlib/newlib/libc/reent/readr.c:49: undefined reference to `_read'
[CFG  ]    /home/XXXXXX/tmp/ctng/x-tools-067-moxie-unknown-moxiebox/moxie-unknown-moxiebox/lib/gcc/moxie-unknown-moxiebox/13.2.0/../../../../moxie-unknown-moxiebox/bin/ld: /home/XXXXXX/tmp/ctng/x-tools-067-moxie-unknown-moxiebox/moxie-unknown-moxiebox/moxie-unknown-moxiebox/lib/libc.a(libc_a-writer.o): in function `_write_r':
[CFG  ]    /home/XXXXXX/tmp/ctng/build-067-moxie-unknown-moxiebox/.build/moxie-unknown-moxiebox/src/newlib/newlib/libc/reent/writer.c:49: undefined reference to `_write'
[CFG  ]    /home/XXXXXX/tmp/ctng/x-tools-067-moxie-unknown-moxiebox/moxie-unknown-moxiebox/lib/gcc/moxie-unknown-moxiebox/13.2.0/../../../../moxie-unknown-moxiebox/bin/ld: /home/XXXXXX/tmp/ctng/x-tools-067-moxie-unknown-moxiebox/moxie-unknown-moxiebox/moxie-unknown-moxiebox/lib/libc.a(libc_a-sbrkr.o): in function `_sbrk_r':
[CFG  ]    /home/XXXXXX/tmp/ctng/build-067-moxie-unknown-moxiebox/.build/moxie-unknown-moxiebox/src/newlib/newlib/libc/reent/sbrkr.c:51: undefined reference to `_sbrk'
```

**Comment by cpackham** [comment]
All done.
